modules/misc/xdg-user-dirs.nix

```
@@ -33,55 +33,64 @@ in {
     # https://gitlab.freedesktop.org/xdg/xdg-user-dirs/blob/master/man/user-dirs.dirs.xml
 
     desktop = mkOption {
-      type = types.str;
+      type = with types; either str path;
+      apply = toString;
       default = "$HOME/Desktop";
       description = "The Desktop directory.";
     };
 
     documents = mkOption {
-      type = types.str;
+      type = with types; either str path;
+      apply = toString;
       default = "$HOME/Documents";
       description = "The Documents directory.";
     };
 
     download = mkOption {
-      type = types.str;
+      type = with types; either str path;
+      apply = toString;
       default = "$HOME/Downloads";
       description = "The Downloads directory.";
     };
 
     music = mkOption {
-      type = types.str;
+      type = with types; either str path;
+      apply = toString;
       default = "$HOME/Music";
       description = "The Music directory.";
     };
 
     pictures = mkOption {
-      type = types.str;
+      type = with types; either str path;
+      apply = toString;
       default = "$HOME/Pictures";
       description = "The Pictures directory.";
     };
 
     publicShare = mkOption {
-      type = types.str;
+      type = with types; either str path;
+      apply = toString;
       default = "$HOME/Public";
       description = "The Public share directory.";
     };
 
     templates = mkOption {
-      type = types.str;
+      type = with types; either str path;
+      apply = toString;
       default = "$HOME/Templates";
       description = "The Templates directory.";
     };
 
     videos = mkOption {
-      type = types.str;
+      type = with types; either str path;
+      apply = toString;
       default = "$HOME/Videos";
       description = "The Videos directory.";
     };
 
     extraConfig = mkOption {
-      type = with types; attrsOf str;
+      type = with types; attrsOf (either str path);
+      apply = builtins.mapAttrs (_: toString);
       default = { };
       example = { XDG_MISC_DIR = "$HOME/Misc"; };
```

let's fix this while you are at it already

```suggestion
      example = literalExpression ''
        {
          XDG_MISC_DIR = "$HOME/Misc";
        }
      '';
```